guide/popular-topics/permissions.md

```diff
@@ -0,0 +1,313 @@
+// output: false
+
+const adminPermissions = new Permissions('ADMINISTRATOR');
+console.log(adminperms.has('MANAGE_CHANNELS'));
+// output: true
+console.log(adminperms.has('MANAGE_CHANNELS'), true);
+// output: true
+console.log(adminperms.has('MANAGE_CHANNELS'), false);
+// output: false
+```
+
+### Manipulating permissions
+
+The Permissions object enables you to easily add or remove certain permissions from an existing bit field without having to worry about bitwise operations. Both `.add()` and `.remove()` can take a single permission flag or number, an array of permission flags or numbers or multiple permission flags or numbers as multiple parameters.
+
+```js
+const { Permissions } = require('discord.js');
+
+const permissions = new Permissions(268550160);
+console.log(permissions.has('KICK_MEMBERS'));
+// output: false
+permissions.add('KICK_MEMBERS');
+console.log(permissions.has('KICK_MEMBERS'));
+// output: true
+permissions.remove(2);
+console.log(permissions.has('KICK_MEMBERS'));
+// output : false
+```
+
+You can utilize these methods to adapt permissions or overwrites without touching the other flags. To achieve this you can get the existing permissions for a role, manipulating the bit field as described above and passing the changed bit field to `role.setPermissions()`.
+
+<tip>In the stable branch `role.permissions` returns a number which needs to be converted to a Permissions object for this to work as described here, we covered how to achieve this in the section "Converting permission numbers to Objects"</tip>
+
+## Final permissions
+
+Discord.js features two utility functions to easily determine the final permissions for a guild member or role in a specific channel: `.permissionsFor()` on the [GuildChannel](https://discord.js.org/#/docs/main/stable/class/GuildChannel?scrollTo=permissionsFor) class and `.permissionsIn()` on the [GuildMember](https://discord.js.org/#/docs/main/stable/class/GuildMember?scrollTo=permissionsIn) class (Note that the Role class does not yet feature a `.permissionsIn()` function on stable). Both return a Permissions object.
+
+To check your bots permissions in the channel the command was used in you could use something like this:
+
+```js
+// Final permissions for a guild member using permissionsFor
+if (message.channel.permissionsFor(message.guild.me).has(['SEND_MESSAGES', 'VIEW_CHANNEL'])) {
+	return console.log('I can send a message here');
+}
+
+// Final permissions for a guild member using permissionsIn
+
+if (message.guild.me.permissionsIn(message.channel).has(['SEND_MESSAGES', 'VIEW_CHANNEL'])) {
+	return console.log('I can send a message here');
+}
+
+// Final permissions for a role
+if (channel.permissionsFor(myRole).has(['SEND_MESSAGES', 'VIEW_CHANNEL'])) {
+	channel.send('This role can send a message here');
+}
+```
+As you will generally want to allow `ADMINISTRATOR` to override you don't need to specify a second parameter here. It defaults to true.
```

Comment:
This line is the opposite of what it needs to be. permissionsFor will return Permissions.all if the member has an Administrator Role, however, you should never checkAdmin from the return of a permissionsFor, because the administrator permission does not apply from channel overwrites.

The above examples should also reflect has(permissions, false) / missing(permissions, false) ect.

Reply:
After more testing and major brain damage from overthinking: That line should not exist at all neither this nor the other way.
Since permissionsFor() returns Permissions.all on global ADMINISTRATOR there is no way to check explicit permissions on the object returned by that method.

Instead added a warning to clarify that.
Further shuffled and restructured the guide a little in the next commit to get a better logical reading flow